src/types/core/pool.ts

```diff
@@ -241,3 +242,21 @@ export function getAssetsOrder(pool: Pool, assetInfo: AssetInfo) {
 		return [pool.assets[1], pool.assets[0]] as Array<Asset>;
 	}
 }
+
+/**
+ * Function to remove pools that are not used in paths.
+ * @param pools Array of Pool types to check for filtering.
+ * @param paths Array of Path types to check the pools against.
+ * @returns Filtered array of Pools.
+ */
+export function removedUnusedPools(pools: Array<Pool>, paths: Array<Path>): Array<Pool> {
```

Comment:
Code looks a little messy. Consider using the ".filter" function to filter the code inplace.
`const filteredPools: Set<Pool> = new Set(
  pools.filter(pool => paths.some(path => path.pools.some(pathPool => pathPool.address === pool.address)))
);
`

Reply:
Let me check if that does the job, no idea that exists in TS